dvc/objects/db/base.py

```diff
@@ -30,6 +35,8 @@ def __init__(self, fs, path_info, **config):
         self.cache_type_confirmed = False
         self.slow_link_warning = config.get("slow_link_warning", True)
         self.tmp_dir = config.get("tmp_dir")
+        self.read_only = config.get("read_only", False)
+        self.staging: Optional["ObjectDB"] = config.get("staging", None)
```

Comment:
Hm, could you elaborate why we need to know if it is a staging db in db itself and not just on a higher level, like `repo.odb.staging`? Is it because you could have two odbs (memfs and localfs), or maybe also some other reason?

Reply:
Ah, so this is kinda like "my staging odb" from the odb's perspective. So this is mini-version of simple mirrors? :slightly_smiling_face: If so, it still feels like high-level things leaking into base odb logic. If we still need this to be this way, maybe it should be some kind of wrapper?

Reply:
Yes it's more like a simple mirror from the ODB's perspective.

I did it this way because it seems like whether an tree is staged or actually saved to `repo.odb.local`, I should be able to use any ODB method other than linking with `repo.odb.local`, and the ODB should handle whether the tree object gets loaded from our "actual" cache or from the temp staging area.

Reply:
Ah, I see. It still seems weird to have one embedded into another like that. Like, there is one staging db per project (or two with mem and local?), but here we have a mechanism that assumes that any odb might have a staging odb inside. Trying to think if there is a way to put this high level logic above db class and not inside of it...